source/beamline/AMDetectorTriggerSource.cpp

```diff
@@ -40,13 +43,56 @@ void AMDetectorTriggerSource::setFailed(){
 	emit failed();
 }
 
- AMDetectorDwellTimeSource::~AMDetectorDwellTimeSource(){}
+AMArmedDetectorTriggerSource::AMArmedDetectorTriggerSource(const QString &name, QObject *parent) :
+	AMDetectorTriggerSource(name, parent)
+{
+	detectorArmingMapper_ = new QSignalMapper(this);
+}
+
+AMArmedDetectorTriggerSource::~AMArmedDetectorTriggerSource()
+{
+}
+
+void AMArmedDetectorTriggerSource::trigger(AMDetectorDefinitions::ReadMode readMode)
+{
+	readMode_ = readMode;
+	armedDetectors_.clear();
+	for(int x = 0, size = triggerSourceDetectors_.count(); x < size; x++)
+		detectorArmingMapper_->setMapping(triggerSourceDetectors_.at(x), triggerSourceDetectors_.at(x));
+	connect(detectorArmingMapper_, SIGNAL(mapped(QObject*)), this, SLOT(onDetectorArmed(QObject*)));
+
+	for(int x = 0, size = triggerSourceDetectors_.count(); x < size; x++)
+		triggerSourceDetectors_.at(x)->arm();
+}
+
+void AMArmedDetectorTriggerSource::addDetector(AMDetector *detector)
+{
+	triggerSourceDetectors_.append(detector);
+	connect(detector, SIGNAL(armed()), detectorArmingMapper_, SLOT(map(QObject*)));
```

Comment:
Silly question @davidChevrier: is it possible for a detector to be armed before trigger(...) is called? In this case, the armed() signal could potentially be emitted before there is a mapping for it.